File: src/main/java/dev/strawn/frugal/FrugalTokenTypes.java
package dev.strawn.frugal;

import com.intellij.psi.TokenType;
import com.intellij.psi.tree.IElementType;

public class FrugalTokenTypes {
    public static final IElementType WHITE_SPACE = TokenType.WHITE_SPACE;
    public static final IElementType IDENTIFIER = new IElementType("IDENTIFIER", FrugalLanguage.INSTANCE);
    public static final IElementType NUMBER = new IElementType("NUMBER", FrugalLanguage.INSTANCE);
    public static final IElementType STRING = new IElementType("STRING", FrugalLanguage.INSTANCE);
    public static final IElementType COMMENT = new IElementType("COMMENT", FrugalLanguage.INSTANCE);
    public static final IElementType SYMBOL = new IElementType("SYMBOL", FrugalLanguage.INSTANCE);
    
    // Keywords
    public static final IElementType NAMESPACE = new IElementType("NAMESPACE", FrugalLanguage.INSTANCE);
    public static final IElementType INCLUDE = new IElementType("INCLUDE", FrugalLanguage.INSTANCE);
    public static final IElementType SERVICE = new IElementType("SERVICE", FrugalLanguage.INSTANCE);
    public static final IElementType STRUCT = new IElementType("STRUCT", FrugalLanguage.INSTANCE);
    public static final IElementType ENUM = new IElementType("ENUM", FrugalLanguage.INSTANCE);
    public static final IElementType EXCEPTION = new IElementType("EXCEPTION", FrugalLanguage.INSTANCE);
    public static final IElementType UNION = new IElementType("UNION", FrugalLanguage.INSTANCE);
    public static final IElementType TYPEDEF = new IElementType("TYPEDEF", FrugalLanguage.INSTANCE);
}

File: src/main/kotlin/dev/strawn/frugal/FrugalTokenTypes.kt
package dev.strawn.frugal

import com.intellij.psi.TokenType
import com.intellij.psi.tree.IElementType

object FrugalTokenTypes {
    @JvmField val WHITE_SPACE = TokenType.WHITE_SPACE
    @JvmField val IDENTIFIER = IElementType("IDENTIFIER", FrugalLanguage)
    @JvmField val NUMBER = IElementType("NUMBER", FrugalLanguage)
    @JvmField val STRING = IElementType("STRING", FrugalLanguage)
    @JvmField val COMMENT = IElementType("COMMENT", FrugalLanguage)
    @JvmField val SYMBOL = IElementType("SYMBOL", FrugalLanguage)
    
    // Keywords
    @JvmField val NAMESPACE = IElementType("NAMESPACE", FrugalLanguage)
    @JvmField val INCLUDE = IElementType("INCLUDE", FrugalLanguage)
    @JvmField val SERVICE = IElementType("SERVICE", FrugalLanguage)
    @JvmField val STRUCT = IElementType("STRUCT", FrugalLanguage)
    @JvmField val ENUM = IElementType("ENUM", FrugalLanguage)
    @JvmField val EXCEPTION = IElementType("EXCEPTION", FrugalLanguage)
    @JvmField val UNION = IElementType("UNION", FrugalLanguage)
    @JvmField val TYPEDEF = IElementType("TYPEDEF", FrugalLanguage)
}